src/System.Private.ServiceModel/src/System/IdentityModel/Selectors/X509SecurityTokenProvider.cs

```diff
@@ -20,7 +20,11 @@ public X509SecurityTokenProvider(X509Certificate2 certificate)
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("certificate");
             }
 
+#if TARGETS_WINDOWS 
```

Comment:
I would prefer to have an overload on X509SercurityTokenProvider constructor to specify whether to clone or not. This can then be passed along in the GetTokenCoreAsync call where a new X509SecurityToken is constructed as that already has an overload for whether to clone or not. That way we can avoid as many `#if` blocks as possible. We already have most of the functionality needed throughout this code to achieve this with minimal changes and minimal platform specific code.